crates/host/src/oci.rs

```diff
@@ -17,34 +18,32 @@ use tokio::fs;
 use tokio::io::AsyncWriteExt;
 use wascap::jwt;
 
-const OCI_VAR_REGISTRY: &str = "OCI_REGISTRY";
-const OCI_VAR_USER: &str = "OCI_REGISTRY_USER";
-const OCI_VAR_PASSWORD: &str = "OCI_REGISTRY_PASSWORD";
-
 const PROVIDER_ARCHIVE_MEDIA_TYPE: &str = "application/vnd.wasmcloud.provider.archive.layer.v1+par";
 const WASM_MEDIA_TYPE: &str = "application/vnd.module.wasm.content.layer.v1+wasm";
 const OCI_MEDIA_TYPE: &str = "application/vnd.oci.image.layer.v1.tar";
 
 fn determine_auth(
     image_reference: &str,
     creds_override: Option<HashMap<String, String>>,
+    host_oci_opts: &OciOptions,
 ) -> RegistryAuth {
-    if let Some(hm) = creds_override {
-        match (hm.get("username"), hm.get("password")) {
-            (Some(un), Some(pw)) => RegistryAuth::Basic(un.to_string(), pw.to_string()),
-            _ => RegistryAuth::Anonymous,
+    match (
+        host_oci_opts.oci_registry.as_ref(),
+        host_oci_opts.oci_user.as_ref(),
+        host_oci_opts.oci_password.as_ref(),
+    ) {
```

Comment:
nit; you could match on `(creds_override, host_oci_opts)` tuple to avoid the `as_ref()` and the `is_some()` below